### PR TITLE
Implemented cooperations details with buttons

### PR DESCRIPTION
**Pull request by Mav-Ivan**
https://github.com/ita-social-projects/SpaceToStudy-Client/assets/110425368/9a0a1531-3180-4dcd-9c8d-100afb56e1d2

### CODE REVIEW

File: src/containers/my-cooperations/cooperation-details/CooperationDetails.constans.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { ButtonProps } from '@mui/material/Button'
 import TodayIcon from '@mui/icons-material/Today'
 
 import Activities from '~/containers/my-cooperations/cooperation-activities/CooperationActivities'
+import MyCooperationsDetails from '../my-cooperations-details/MyCooperationsDetails'
```
Comment (Tolik170) [inline src/containers/my-cooperations/cooperation-details/CooperationDetails.constans.tsx:6]: use import aliases

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import palette from '~/styles/app-theme/app.pallete'
+import { TypographyVariantEnum } from '~/types'
+import { ellipsisTextStyle } from '~/utils/helper-functions'
+
+export const style = {
+  header: {
+    typography: TypographyVariantEnum.H6,
+    color: palette.basic.blueGray,
+    ml: '20px',
+    mb: '30px'
```
Comment (Tolik170) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts:10]: can you use a single m property or do you need to rewrite these margins?

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+              size={SizeEnum.Medium}
+              sx={style.buttons}
+              variant={ButtonVariantEnum.Tonal}
+            >
+              <PersonIcon />
+              {t('cooperationDetailsPage.profile')}
+            </AppButton>
+          </Box>
+        </Box>
+        <Typography sx={style.titles}>Tutoring subject & level:</Typography>
```
Comment (Tolik170) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:114]: text should have translation

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        <Box sx={style.subjectContainer}>
+          <DesignServicesIcon sx={{ color: offer.category.appearance.color }} />
```
Comment (Tolik170) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:116]: move sx to styles

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+          />
+        </Box>
+        <Typography sx={style.titles}>About cooperation:</Typography>
+        <ShowMoreCollapse
+          collapsedSize={30}
+          collapsedTextLength={100}
+          description={offer.description}
+          sx={style.aboutCooperation}
+          title={''}
+          withoutTitle
```
Comment (Tolik170) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:131]: do you need to pass title and withoutTitle props together?

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        />
+        <Typography sx={style.titles}>Tutoring languages:</Typography>
+        <Box sx={style.languageContainer}>{languages}</Box>
+        <Typography sx={style.titles}>Pricing:</Typography>
+        <Typography>{`${price} UAH/hour`}</Typography>
```
Comment (Tolik170) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:136]: translations for languages and pricing

File: src/types/my-cooperations/interfaces/myCooperations.interfaces.ts [hunk 7]
```diff
@@ -49,3 +49,7 @@ export interface AdditionalPropsInterface {
   t: TFunction
   navigate: NavigateFunction
 }
+
+export interface GetCooperationByIdParams {
```
Comment (Tolik170) [inline src/types/my-cooperations/interfaces/myCooperations.interfaces.ts:53]: maybe just GetCooperationParams

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    gridTemplateColumns: 'repeat(2, max-content)',
+    gap: '40px',
+    padding: '20px'
+  },
+  profileContainer: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: '10px',
+    pt: '0px',
+    pl: '0px',
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts:30]: you have `padding`, `pt`, `pl`
try to write it in one property `p`

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  const { id = '' } = useParams()
+  const navigate = useNavigate()
+  const { setChatInfo } = useChatContext()
+
+  const getDetails = useCallback(
+    () => cooperationService.getCooperationById(id),
+    [id]
+  )
+
+  const { response, loading } = useAxios({
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:38]: we add some names for response.
here better use rewriting it `response: detailsResponse` and also can make it for `loading`

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    service: getDetails,
+    defaultResponse: defaultOffer
+  })
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:41]: as Markiyan said. Think about if we have empty response and we should handle it

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.constants.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
+export const defaultOffer = {
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.constants.tsx:1]: we already have `defaultResponse` const. can reuse it and not to create a new variable for it

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+  const { setChatInfo } = useChatContext()
+
+  const getDetails = useCallback(
+    () => cooperationService.getCooperationById(id),
+    [id]
+  )
+
+  const { response, loading } = useAxios({
+    service: getDetails,
+    defaultResponse: defaultOffer
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:40]: use existing `defaultResponse`

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+  return loading ? (
+    <Loader size={50} />
+  ) : (
+    <Box>
+      <Typography sx={style.header}>
+        {t('cooperationDetailsPage.details')}
+      </Typography>
+
+      <Box sx={style.container}>
+        <Typography sx={style.titles}>Cooperations title:</Typography>
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:75]: translations here

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+        <Typography sx={style.title}>{offer.title}</Typography>
+        <Typography sx={style.titles}>Tutor:</Typography>
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:77]: the same

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+        <Box sx={style.subjectContainer}>
+          <DesignServicesIcon sx={{ color: offer.category.appearance.color }} />
+          <Typography>{offer.category.name}</Typography>
+          <SubjectLevelChips
+            color={offer.category.appearance.color}
+            proficiencyLevel={offer.proficiencyLevel}
+            subject={offer.subject.name}
+          />
+        </Box>
+        <Typography sx={style.titles}>About cooperation:</Typography>
```
Comment (dmtrth25) [inline src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:124]: translations here

File: src/containers/my-cooperations/cooperation-details/CooperationDetails.constans.tsx (hunk 1 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/cooperation-details/CooperationDetails.constans.tsx]: done

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts (hunk 2 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts]: yes, done

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 4 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: done

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 5 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx:149]: no i fixed that

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 6 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: done

File: src/types/my-cooperations/interfaces/myCooperations.interfaces.ts (hunk 7 above)
Comment (Mav-Ivan) [reply src/types/my-cooperations/interfaces/myCooperations.interfaces.ts]: agree

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts (hunk 8 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.styles.ts]: done

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 10 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: we need to discuss it

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 12 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: i changed a bit

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 14 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: done

File: src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx (hunk 15 above)
Comment (Mav-Ivan) [reply src/containers/my-cooperations/my-cooperations-details/MyCooperationsDetails.tsx]: done

**Review by Marichka0406 — APPROVED** [review]

**Review by Tolik170 — APPROVED** [review]

**Review by Mygal-Alex — APPROVED** [review]

**Pull request merged**